sockeye/training.py

```diff
@@ -135,17 +185,73 @@ def sym_gen(seq_lens):
                                                            source_embed_length,
                                                            source_embed_seq_len)
 
-            # decoder
-            # target_decoded: (batch-size, target_len, decoder_depth)
-            target_decoded = self.decoder.decode_sequence(source_encoded, source_encoded_length, source_encoded_seq_len,
+            # target embedding
+            (target_embed,
+             target_embed_length,
+             target_embed_seq_len) = self.embedding_target.encode(target, target_length, target_seq_len)
+
+            if self.config.scheduled_sampling_type is not None:
```

Comment:
it would be nice to move this into its own function to avoid the large if clause